src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/Subscriptions/DependencySharedProjectsSubscriber.cs

```diff
@@ -110,30 +111,28 @@ private void SubscribeToConfiguredProject(IProjectSubscriptionService subscripti
 
             EnsureInitialized();
 
-
-            await _tasksService.LoadedProjectAsync(async () =>
+            await _tasksService.LoadedProjectAsync(() =>
             {
                 if (_tasksService.UnloadCancellationToken.IsCancellationRequested)
                 {
-                    return;
+                    return Task.CompletedTask;
```

Comment:
The pattern here is usually to ThrowIfCancellationRequested and to fault the block - that's exactly what LoadProjectAsync throws if the project is already unloaded at this point.

Reply:
Updated.

Looking at the implementation of `LoadProjectAsync`, it seems to do this check synchronously before calling our func anyway. No harm in doing it again here though.